lib/mix/lib/mix/tasks/escript.build.ex

```diff
@@ -242,6 +247,19 @@ defmodule Mix.Tasks.Escript.Build do
     end)
   end
 
+  defp strip_beams(tuples) do
+    for {basename, maybe_beam} <- tuples do
```

Comment:
Since we are traversing all of them, I would recommend doing this:

```elixir
for {basename, maybe_beam} <- tuples, Path.extname(basename) == ".beam", do:  {basename, strip_beam(maybe_beam)}
```

The `strip_beam/1` no longer needs a case clause.